### PR TITLE
Add a tutorial about resilient aliases

### PR DESCRIPTION
**Pull request by nicolas-rabault**
⚠ PLEASE DO NOT DELETE THE TEXT BELOW ⚠

## Pull request's description

*Add here a description of the modifications you made.*

## Checklist before merging (please do not edit)
You must review you work before to submit the review to others.

### Self-review of your content
Remember the content must be readable and understandable by someone else than yourself.
- [ ] From a technical point of view.
- [ ] From a grammatical point of view (spelling mistakes, typos, clarity, etc.), and following the guidelines at the bottom.

### External reviews of your content
- [ ] You requested a technical review.
- [ ] You requested a grammatical review.
- [ ] You validated with @K0rdan or @alexgeron-Luos that it is safe to merge.

### Some guidelines to keep in mind
- Colons (:), semi-colons (;), exclamation (!), and interrogation points (?) are not preceded by a space (like full stops and commas). E.g.: Colons!
- File names and/or address are put in *italic*. E.g.: The file _main.c_.
- Functions, variables, or more generally short codes are put between grave accents. E.g.: To obtain `code()`, type \`code()\`.
- Long codes are put into blocks of code with three grave accent on each side, and the language's name:<br />
\`\`\`c<br />
// Some C language <br />
\`\`\`<br />

- "Luos engine" has a upper case on the **L** for "Luos" and lower case on the **e** for "engine".
- We call it "Luos engine" as a proper name, and ***not*** "the Luos engine".
- Following that fashion, anything that's owned by "Luos engine" implies that we must use `'s` as the standard English rule to show the possessive case, e.g. "Luos engine's API".
- The names pipe, gate, inspector, sniffer, app or application, driver, etc. have a lower case and can be refereed with the determiner `the`. E.g.: The inspector.

### CODE REVIEW

**Review by Simonbdy — CHANGES_REQUESTED** [review]
Very nice and clear.

Here are my comments:
- Two "TODO" are to be replaced at the beginning (19) and at the end (177)
- The function `device.custom_led.rename("custom led")` in line 85 was replaced by `device.led.rename("custom led")`
- The word "somewhere" is too vague, added examples.

**Review by Simonbdy — APPROVED** [review]

File: tutorials/resilient-alias/resilient-alias.mdx [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+---
+
+import Image from '@site/src/components/Images.js';
+
+
+# Resilient aliases management
+
+Luos allows you to deal with service aliases. Aliases are much more practical than ids and allow developers to naturally find and access a specific service somewhere in the network.
+
+In some developments, you will need to duplicate the same service multiple times. For example, if you are making a robot with multiple motors you could have the exact same board running the exact same firmware at multiple times and at different places on your device.
```
Comment (BiliouriV) [inline tutorials/resilient-alias/resilient-alias.mdx:12]: running the exact same firmware multiple times

File: tutorials/resilient-alias/resilient-alias.mdx [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+:::info
+You can find an example of resilient aliases management here: TODO.
+:::
+
+# How aliases are managed
+
+Every service needs an alias. Basically, when you create your service you will have to define your service default's alias.
+
+After this creation, your alias may be modified in different ways.
```
Comment (BiliouriV) [inline tutorials/resilient-alias/resilient-alias.mdx:26]: can be modified

File: tutorials/resilient-alias/resilient-alias.mdx [hunk 3]
```diff
@@ -0,0 +1,178 @@
...
+
+Luos provides a flexible solution allowing you to optionally save your aliases in the best way possible for your project.
+
+Let's see how to make it!
+
+## Default aliases
+
+In every case, you need to define a default alias at the service creation. This is mandatory, and this alias will be used as a backup if your custom alias is removed.
+
+Let's use a simple les service as an example. In the package inititalization, you need to create your service with the associated alias:
```
Comment (BiliouriV) [inline tutorials/resilient-alias/resilient-alias.mdx:48]: led*

File: tutorials/resilient-alias/resilient-alias.mdx [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+```
+
+## How to save aliases for multiple packages
+
+The main concept of Luos is to keep services and packages loosely coupled. This means that your package should have everything needed locally and does not depend on any code somewhere else.
+
+If you have multiple packages on a node and if you want to be able to save your aliases on all those packages, you may want to have a common function used by all the packages to store aliases. In this case, we advise you to consider your memory management functions as a library and to make your packages dependent on this library.
+
+:::info
+You can find an example of resilient aliases management as a library here: TODO
```
Comment (BiliouriV) [inline tutorials/resilient-alias/resilient-alias.mdx:177]: dont forget this also

**Review by BiliouriV — APPROVED** [review]
Nice and clean work. Bravo!

**Review by JeromeGalan — COMMENTED** [review]
Good job !

I think, there is too much text to read before the "saving aliases" paragraph.
Maybe we should add some pictures ?

For example :

- After "Luos will add a number at the end of your alias depending on their position from the node performing the detection. This guarantees aliases unicity", we could add a routing table with :
Motor_1  /  Motor_2  /  Motor_3


- After "and to choose a practical alias name such as “left_motor” or “front_sensor”", we could add this kind of pic  (mine is ugly :\\/ ...) :

sensor_10    ---- rename ---->    front_sensor
motor_4    ---- rename ---->    left_motor

**Review by Simonbdy — APPROVED** [review]
ok for me

**Review by lorenzi-nicolas — APPROVED** [review]
Ok for me, the thumbnail has been added

**Review by JeromeGalan — APPROVED** [review]

**Pull request merged**